### ISSUE TITLE
Possibility to turn this into a reusable plugin with semantic-release (core)?

### ISSUE THREAD

**Issue by manuschillerdev** [semantic-release-plus/semantic-release#294]
## New feature motivation
In my understanding, `semantic-release-plus` always needs to catch up with `semantic-release` when the upstream version changes. A reusable plugin for filtering commits by path in monorepos would probably be a very robust solution for a common problem. Would that be feasible?

## New feature description
The possibility to filter commits by path is one of the core features of `semantic-release-plus`. If it were possible to extract the functionality this would be a super cool feature for the `semantic-release` ecosystem.

## New feature implementation
Extract the functionality to filter relevant commits by path into a reusable plugin with `semantic-release`.
I am not sure if this is possible at all or what would need to be done implementation-wise.

**Comment by JoA-MoS** [comment]
I agree, and it's been almost a year since you wrote this, Initially I tried very hard to get the upstream semantic-release to incorporate my changes but alas they were not interested in supporting a monorepo, and at the time the structure of the code with plugins did not provide a way to filter commits in a plugin. There are really only 2 parts that would need to be extracted and integrated into the core of semantic-release

1. [The filter mechanism](https://github.com/semantic-release-plus/semantic-release-plus/blob/alpha/packages/srp/src/lib/git.js#L56C37-L56C42)
2. [The generating and retrieving git notes needs to be specific per release](https://github.com/semantic-release-plus/semantic-release-plus/blob/alpha/packages/srp/src/lib/git.js#L366C26-L366C37)

The filter Mechanism is in this repo but I think 2 is only in my alpha branch which is located at [semantic-release-plus/semantic-release-plus](https://github.com/semantic-release-plus/semantic-release-plus/tree/alpha)

My current job has started using semantic-release-plus so I will start making updates. It might be worth the effort to see if the upstream mentality has changed but previously they were not interested in the monorepo support.

**Comment by kf6kjg** [comment]
Trying to restart the discussion: https://github.com/semantic-release/semantic-release/issues/1688#issuecomment-2197390178
